### PR TITLE
Confusion Matrix

### PR DESCRIPTION
**Pull request by t-wangjohn**
Confusion Matrix added in model assessment for multiclass and binary classification

## Description

binary classification (light mode)
MicrosoftTeams-image

binary classification (dark mode)
MicrosoftTeams-image (1)

multiclass classification (light mode)
MicrosoftTeams-image (2)

multiclass classification (dark mode)
MicrosoftTeams-image (3)

Too few classes selected error (there's another one with at most 20, but I don't have a dataset handy that triggers the error):
MicrosoftTeams-image (4)

 

## Checklist

- [x] I have added screenshots above for all UI changes.
- [x] I have added e2e tests for all UI changes.
- [x] Documentation was updated if it was needed.

### CODE REVIEW

**Comment by t-wangjohn** [comment]
image

**Review by romanlutz — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ConfusionMatrixHeatmap.tsx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+              styles={FluentUIStyles.limitedSizeMenuDropdown}
+            />
+          </StackItem>
+        </Stack>
+        {this.state.selectedClasses.length >= this.minDisplayableClasses &&
+          this.state.selectedClasses.length <= this.maxDisplayableClasses && (
+            <StackItem className={classNames.chart}>
+              <HeatmapHighChart
+                id="ModelOverviewConfusionMatrix"
+                configOverride={{
```
Comment (tongyu-microsoft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ConfusionMatrixHeatmap.tsx:181]: looks like there is some issue with the global eslint rule. This file is over 300 lines but there is no lint error in the test. Can we move this configOverride in a separate file to reduce lines? I remember previously the max-line rule in eslintrc file in each folder works and pops up errors if any.

**Review by tongyu-microsoft — APPROVED** [review]

**Comment by tongyu-microsoft** [comment]
Could you confirm that it looks good for large screens?

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ConfusionMatrixHeatmap.tsx (hunk 1 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ConfusionMatrixHeatmap.tsx:181]: We exempted the entirety of `model-assessment` so it's a consequence of that. I'll create a PR to undo that and exempt just specific files, so at least for new files it won't apply! I'll also reduce this file's size.

**Review by JingyaChen — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverviewChartPivot.tsx [hunk 2]
```diff
@@ -178,6 +179,17 @@ export class ModelOverviewChartPivot extends React.Component<
             selectedMetric={this.state.selectedMetric}
           />
         </PivotItem>
+        {(this.context.modelMetadata.modelType === ModelTypes.Binary ||
```
Comment (gaugup) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverviewChartPivot.tsx:182]: Can we disable the confusion matrix logic for large data with the function `IfEnableLargeData()`? We don't support confusion matrix in SDK backend as yet.

Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverviewChartPivot.tsx:182]: Sure. Can you create a work item to track it @gaugup ?

Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverviewChartPivot.tsx]: Added the extra check

**Pull request merged**